src/sentry/web/frontend/organization_auth_settings.py

```diff
@@ -83,42 +79,32 @@ class AuthProviderSettingsForm(forms.Form):
     return form
 
 
-@region_silo_view
-class OrganizationAuthSettingsView(OrganizationView):
+@control_silo_view
+class OrganizationAuthSettingsView(ControlSiloOrganizationView):
     # We restrict auth settings to org:write as it allows a non-owner to
     # escalate members to own by disabling the default role.
     required_scope = "org:write"
 
     def _disable_provider(
         self, request: Request, organization: RpcOrganization, auth_provider: RpcAuthProvider
     ):
-        with outbox_context(transaction.atomic(router.db_for_write(OrganizationMember))):
-            self.create_audit_entry(
-                request,
-                organization=organization,
-                target_object=auth_provider.id,
-                event=audit_log.get_event_id("SSO_DISABLE"),
-                data=auth_provider.get_audit_log_data(),
-            )
-
-            OrganizationMember.objects.filter(organization_id=organization.id).update(
-                flags=F("flags")
-                .bitand(~OrganizationMember.flags["sso:linked"])
-                .bitand(~OrganizationMember.flags["sso:invalid"])
-            )
-
-            RegionOutbox(
-                shard_scope=OutboxScope.ORGANIZATION_SCOPE,
-                shard_identifier=organization.id,
-                category=OutboxCategory.DISABLE_AUTH_PROVIDER,
-                object_identifier=auth_provider.id,
-            ).save()
```

Comment:
@GabeVillalobos This is how we were removing authprovider replica data before. The handler for this outbox would remove the replica https://github.com/getsentry/sentry/blob/8f47c4cb5ae62e3ee539f0f94ff497bf4ab11b79/src/sentry/receivers/outbox/region.py#L76-L79